src/g_knob.c

```diff
@@ -0,0 +1,1048 @@
+/* Copyright (c) 2020 Antoine Rousseau.
+ * For information on usage and redistribution, and for a DISCLAIMER OF ALL
+ * WARRANTIES, see the file, "LICENSE.txt," in this distribution. */
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <ctype.h>
+#include "m_pd.h"
+#include "g_canvas.h"
+
+#include "g_all_guis.h"
+#include <math.h>
+
+#ifdef _MSC_VER
+#define snprintf _snprintf
+#endif
+
+#ifdef _WIN32
+#include <io.h>
+#else
+#include <unistd.h>
+#endif
+
+#ifndef M_PI
+    #define M_PI 3.14159265358979323846
+#endif
+
+#define MKNOB_DEFAULT_SENSITIVITY 128
```

Comment:
I think these can probably drop the "MKNOB_" prefix as they are only used internally or maybe becomes simply "KNOB_". I notice the other item guis have similar local defines, but without prefixes.

Reply:
urgh... right! fixing...

Reply:
i removed the prefixes.